presto-parquet/src/main/java/com/facebook/presto/parquet/writer/valuewriter/TimestampValueWriter.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.writer.valuewriter;
+
+import com.facebook.presto.common.block.Block;
+import com.facebook.presto.common.type.Type;
+import org.apache.parquet.column.values.ValuesWriter;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class TimestampValueWriter
```

Comment:
Should we write a test for this?

Reply:
Thanks @tdcmeehan. I thought the existing functional tests are enough. None of these writer classes have individual tests. Should we write one for this?

Reply:
Verified that the TestParquetReader and TestHiveFileFormats::testOptimizedParquetWriter does round trip tests of this writer.

Reply:
I'm guessing this fixes a bug in the writing of timestamp values as well.  Is there any way this bug could be captured in a test that fails without the patch above, and succeeds with the patch above?

Reply:
There is no way to avoid the patch and reproduce the bug.